### DISCUSSION TITLE
Self hosting

### DISCUSSION THREAD

**Discussion by NitroNine** [rekkyrosso/ampcast#21] [Q&A] [answered]
Why when I self host does it ask for dev information to link to Apple Music and Spotify, when the ampcast.app does not?

**Comment by rekkyrosso** [comment]
I don't distribute any keys with the self-hosted option.

It's pretty easy to get a Spotify client id but an Apple Music license costs money.

**Reply by NitroNine** [reply]
Yeah I have spotify keys, but it just brings me back to ampcast and tells me to login to spotify but doesn't do anything?

**Reply by rekkyrosso** [reply]
Make sure to add the correct callback URL and select the right APIs when you register your app with Spotify.

I just tested this process and it worked fine.

You can also check ampcast's logs to see if you get any login errors. Checking the browser's dev console might also help.

Settings>Advanced>Logs

**Reply by NitroNine** [reply]
Yep, checked all of that.  Plex also brings me to the login to plex, logs in, then just shows a blank page with /auth/plex/callback/ at the end of my IP.  ampcast doesn't log in.

btw, when I try your app link, plex works fine.

**Reply by rekkyrosso** [reply]
Are you using a custom name domain or https?

**Reply by rekkyrosso** [reply, marked as answer]
Ampcast version 0.9.13 should fix any login problems you have.
